intake/container/serializer.py

```diff
@@ -54,7 +63,10 @@ def encode(self, obj, container):
         if container in ['ndarray', 'xarray'] and msgpack_numpy:
             return msgpack.packb(obj, **np_pack_kwargs)
         elif container == 'dataframe':
-            return obj.to_msgpack()
+            pa = check_pyarrow()
+
+            context = pa.default_serialization_context()
+            return memoryview(context.serialize(obj).to_buffer())
```

Comment:
Does `memoryview` do anything here?

Reply:
See https://github.com/intake/intake/pull/462#discussion_r377119345. It's a workaround for

```python
In [5]: a = pa.array([1, 2, 3])

In [9]: msgpack.packb(a.buffers()[0])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-08194270ba6f> in <module>
----> 1 msgpack.packb(a.buffers()[0])

~/miniconda3/envs/pyarrow=0.16.0/lib/python3.8/site-packages/msgpack/__init__.py in packb(o, **kwargs)
     44     See :class:`Packer` for options.
     45     """
---> 46     return Packer(**kwargs).pack(o)
     47
     48

msgpack/_packer.pyx in msgpack._cmsgpack.Packer.pack()

msgpack/_packer.pyx in msgpack._cmsgpack.Packer.pack()

msgpack/_packer.pyx in msgpack._cmsgpack.Packer.pack()

msgpack/_packer.pyx in msgpack._cmsgpack.Packer._pack()

TypeError: can not serialize 'pyarrow.lib.Buffer' object

In [10]: msgpack.packb(memoryview(a.buffers()[0]))
Out[10]: b'\xa1\x07'

```

Reply:
:+1: msgpack's type dispatch is very intolerant of duck types, has lots of `isinstance` checks.